File: wonseok970/BinarySearch/[10815] 숫자 카드.py
n = int(input())
card_list = list(map(int, input().split()))

m = int(input())
check_list = list(map(int, input().split()))

card_list.sort()


def bn_search(start, end, target, array):
    if start > end:
        return False
    mid = (start + end) // 2

    if array[mid] == target:
        return True
    elif array[mid] < target:
        return bn_search(mid + 1, end, target, array)
    else:
        return bn_search(start, mid - 1, target, array)


for i in range(m):
    v = bn_search(0, n - 1, check_list[i], card_list)
    if v:
        print(1, end=" ")
    else:
        print(0, end=" ")

File: gettls/week6/[10815] 숫자 카드.cpp
#include<iostream>
#include<queue>
#include<algorithm>
#pragma warning (disable:4996)
using namespace std;

int N[500000];
int M[500000];
int n, m;
int binarySearch(int target) {
	int start = 0, end = n - 1;
	while (start <= end) {
		int mid = (start + end) / 2;
		if (N[mid] > target) end = mid - 1;
		else if (N[mid] < target) start = mid + 1;
		else if(N[mid] == target) return 1;
	}
	return 0;
}

int main() {
	cin >> n;
	for (int i = 0; i < n; i++) scanf("%d", &N[i]);
	sort(N, N + n);

	cin >> m;
	for (int i = 0; i < m; i++) {
		scanf("%d", &M[i]);
		printf("%d ", binarySearch(M[i]));
	}
}
